bin/webpack.js

```diff
@@ -188,6 +188,10 @@ function processOptions(options) {
 	if(typeof outputOptions.context === "undefined")
 		outputOptions.context = firstOptions.context;
 
+	ifArg("env", function(environment) {
+		outputOptions.env = environment;
```

Comment:
Maybe I'm misunderstanding – but I've updated the check in [`lib/Stats.js`](https://github.com/webpack/webpack/pull/4903/files#diff-baf371f5446dc9bc4b41022587e05b48R206) to handle the bool case(s) that were a result of the preset object. This value (`--env`) seems to be a bit of a special case since the majority of the other properties/values are being read from the `compilation` or defined on the `stats` instance directly. I considered both options but this implementation seemed the least intrusive... I can certainly make a change either way.